core/tx_pool.go

```diff
@@ -339,17 +341,6 @@ func (pool *TxPool) Pending() (map[common.Address]types.Transactions, error) {
 	pool.mu.Lock()
 	defer pool.mu.Unlock()
```

Comment:
Yeah, but since it's using the same lock as `Add`, that means `Pending` contends with `Add`. So it would be good if one of them didn't need the lock. There's so much action around only one mutex.